src/parser/statement.js

```diff
@@ -861,8 +861,22 @@ pp.parseExport = function (node) {
   return this.finishNode(node, "ExportNamedDeclaration");
 };
 
+pp.shouldParseExportDeclaration = function () {
+  return this.state.type.keyword || this.isContextual("async");
+};
+
+pp.isDeclaration = function (node): boolean {
+  return node.type === "VariableDeclaration"
+    || node.type === "FunctionDeclaration"
+    || node.type === "ClassDeclaration";
+};
+
 pp.parseExportDeclaration = function () {
-  return this.parseStatement(true);
+  const statement = this.parseStatement(true);
+  if (!this.isDeclaration(statement)) {
+    this.unexpected(statement.start);
```

Comment:
Since we know what the problem is here (an `export` declaration that doesn't have a valid export declaration or export clause) maybe we can use `this.raise` and hint to the user what the problem was?